irida_sistr_results/irida_api.py

```diff
@@ -107,49 +105,47 @@ def get_sistr_results_for_project(self, project):
 		for sample in sistr_results_for_project:
 			sample_pairs=self.irida_connector.get_resources('/api/samples/'+sample['identifier']+'/pairs')
 
-			if len(sample_pairs) == 0:
-				sistr_info = SampleSistrInfo({'sample': sample,
-						'has_results': False
-						})
-			else:
-				sistr_info = None
+			sistr_info = SampleSistrInfo.create_empty_info(sample)
+
+			if len(sample_pairs) > 0:
 				for sequencing_object in sample_pairs:
 					if (self._has_rel_in_links('analysis/sistr', sequencing_object['links'])):
 						sistr_rel=self._get_rel_from_links('analysis/sistr', sequencing_object['links'])
-	
 						sistr=self.irida_connector.get(sistr_rel)
-						
+
 						if (sistr['analysisState'] != 'COMPLETED'):
-							logger.debug("Skipping automated SISTR results associated with sample="+sample['sampleName']+" as state is not 'COMPLETED'.")
-							if sistr_info is None:
-								sistr_info = SampleSistrInfo({'sample': sample,
-									'paired_files': sequencing_object,
-									'has_results': False
-									})
+							logger.debug(
+								"Skipping automated SISTR results associated with sample=%s as state is not 'COMPLETED'.",
+								sample['identifier'])
 						else:
-							sistr_info_curr=self.get_sistr_info_from_submission(sistr)
-							if (sistr_info is None):
-								sistr_info = sistr_info_curr
-							elif sistr_info_curr.get_qc_status() == 'PASS' and (not sistr_info.has_sistr_results() or sistr_info.get_qc_status() != 'PASS'):
-								sistr_info = sistr_info_curr
-							elif sistr_info_curr.get_qc_status() == 'PASS' and (sistr_info_curr.get_submission_created_date() > sistr_info.get_submission_created_date()):
-								sistr_info = sistr_info_curr
-					elif sistr_info is None:
-						sistr_info = SampleSistrInfo({'sample': sample,
-								'paired_files': sequencing_object,
-								'has_results': False
-								})
+							new_sistr_info = self.get_sistr_info_from_submission(sistr)
+							sistr_info = self._update_sample_sistr_info(sistr_info, new_sistr_info, sistr_workflow_id)
 
 			sistr_results.append(sistr_info)
 		return sistr_results
+
+	def _update_sample_sistr_info(self, curr_sistr_info, new_sistr_info, sistr_workflow_id):
+		if sistr_workflow_id is None or sistr_workflow_id == new_sistr_info.get_submission_workflow_id():
+			if not curr_sistr_info.has_sistr_results():
+				return new_sistr_info
+			elif new_sistr_info.get_qc_status() == 'PASS' and curr_sistr_info.get_qc_status() != 'PASS':
+				return new_sistr_info
+			elif new_sistr_info.get_qc_status() == 'PASS' and (
```

Comment:
These checks only apply to automated analysis submissions. They are meant to avoid the case where you initially upload some good quality fastq files and get a `PASS`, and then upload poor quality fastq files (from a different run) and get a `FAIL`.  Automated analyses don't get re-run for new workflow updates though.

However, if someone wishes too, they could re-run the analysis and share back to the project. These results will get picked up as well, and the most recent set of results will override previous results independent of the quality score (checks done in https://github.com/phac-nml/irida-sistr-results/blob/feature/export-sistr-versions/irida_sistr_results/irida_sistr_results.py#L70).

So, if updating SISTR changes QC from `PASS` to `WARN`, and if a person were to re-run SISTR on some sample, the final output would be the newest SISTR results (with status of `WARN`).

Reply:
Okay, that's great. Thanks for clarifying.